File: app/src/main/java/za/co/riggaroo/datecountdown/repository/EventRepository.java
package za.co.riggaroo.datecountdown.repository;


import android.arch.lifecycle.LiveData;

import java.util.List;

import io.reactivex.Completable;
import za.co.riggaroo.datecountdown.entity.Event;

/**
 * @author rebeccafranks
 * @since 2017/04/21.
 */
public interface EventRepository {

    Completable addEvent(Event event);

    LiveData<List<Event>> getEvents();

    Completable deleteEvent(Event event);


}

File: app/src/main/java/za/co/riggaroo/datecountdown/repository/EventRepository.kt
package za.co.riggaroo.datecountdown.repository


import android.arch.lifecycle.LiveData

import io.reactivex.Completable
import za.co.riggaroo.datecountdown.data.entity.Event

/**
 * @author rebeccafranks
 * @since 2017/04/21.
 */
interface EventRepository {

    fun getEvents(): LiveData<List<Event>>

    fun addEvent(event: Event?): Completable

    fun deleteEvent(event: Event?): Completable

}
